pgcql/pgcql.go

```diff
@@ -0,0 +1,32 @@
+package pgcql
+
+import (
+	"github.com/indexdata/cql-go/cql"
+)
+
+type PgError struct {
+	message string
+}
+
+func (e *PgError) Error() string {
+	return e.message
+}
+
+type Field interface {
+	GetColumn() string
+	SetColumn(column string)
+	Generate(sc cql.SearchClause, queryArgumentIndex int) (string, []any, error)
+}
+
+type Definition interface {
+	AddField(name string, field Field) Definition
+	GetFieldType(name string) Field
+	Parse(q cql.Query, queryArgumentIndex int) (Query, error)
+}
+
+type Query interface {
```

Comment:
Should we use more descriptive names like `PgQuery`, `PgField`? I know it's not needed since this is its own package but it would be useful for the reader.

Reply:
I'd rather not. Just as it's not called `cql.CqlQuery`.